src/main/java/codeu/controller/ParsingEngine.java

```diff
@@ -0,0 +1,30 @@
+package codeu.controller;
+
+import codeu.model.data.BotRequest;
+import codeu.model.data.BotStructure;
+import codeu.model.data.RoutingEngine;
+import java.util.regex.*;
+
+
+public class parsingEngine {
+ 	public void ParseCommands(String message){
+ 		BotRequest request;
```

Comment:
Move this declaration closer to where it is used (i.e. line 23)

Reply:
Also consider having a constructor that completely initializes the request, so this can be made a `final BotRequest request`